### ISSUE TITLE
spaces/special characters in complete project path cause `ERR_INVALID_URL_SCHEME` vite error

### ISSUE THREAD

**Issue by ERIS-VN** [nuxt/framework#9074]
### Environment

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.16.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

![image](https://user-images.githubusercontent.com/69683870/202239590-0795eff9-76ce-41fc-b0d8-8330e4cf4060.png)

### Describe the bug

Error when I use run dev

### Logs

```shell
500
[vite-node] [ERR_INVALID_URL_SCHEME] /node_modules/ufo/dist/index.mjs

at /node_modules/ufo/dist/index.mjs
```

**Comment by therealokoro** [comment]
Same issue with me...

I was about to open an issue if no one had same issue...Thank God someone else does.

I first noticed this issue when I tried upgrading to RC.13.. but then I had to stick with RC.12

**Comment by sdezza** [comment]
same here on:

- Operating System: Windows 10
- Node Version: v18.12.1 (did latest LTS update)
- Nuxt Version: 3.0.0

upgrade and new project, same issue

**Comment by niuwenzheng** [comment]
Same problem.  my path have chinese characters. It is possible to change to a directory that has no Chinese.

**Comment by adamiss5138** [comment]
Same here. I have to stick with rc11. Any other version throws similar error.

**Comment by manniL** [comment]
This seems to be an issue when having "special characters" that have to be URL-encoded in the path.

#8706

**Comment by Kroixyz** [comment]
![image](https://user-images.githubusercontent.com/67718632/202839725-6e17e4a8-a31f-40d1-a863-f3f0438bbfb5.png)

same problem lol

**Comment by belwalpb** [comment]
Same Problome. Even with rc14 as well as stable release.
![image](https://user-images.githubusercontent.com/46915613/202858244-f8899ca0-b0d5-40df-b1f2-0aded50c5a39.png)

**Comment by frasza** [comment]
Same problem here :S

**Comment by manniL** [comment]
> Same problem here :S

Please don't post "+1" or "Same problem".
Instead, upvote issues or comments.

Also see https://github.com/nuxt/framework/issues/9074#issuecomment-1319389101

**Comment by frasza** [comment]
Yes, sorry, you are correct @manniL. Apologies. 
I have dig deeper and what I found is that I had space in my user space folder - "firstname lastname" and when I created new folder without space, it worked.

**Comment by realdinho** [comment]
> Yes, sorry, you are correct @manniL. Apologies. I have dig deeper and what I found is that I had space in my user space folder - "firstname lastname" and when I created new folder without space, it worked, so you are right.

uuufff....finally!!! thanks @frasza 
this is the correct answer:
**The directory of your nuxt app CANNOT CONTAIN A SPACE**

**Comment by Fanreza** [comment]
> Yes, sorry, you are correct @manniL. Apologies. I have dig deeper and what I found is that I had space in my user space folder - "firstname lastname" and when I created new folder without space, it worked, so you are right.

My nuxt folder does not contain any special character, but the parent folder does and it make nuxt error
`some folder -> myprojectfolder -> nuxt-app`

**Comment by Phatza** [comment]
> > Yes, sorry, you are correct @manniL. Apologies. I have dig deeper and what I found is that I had space in my user space folder - "firstname lastname" and when I created new folder without space, it worked, so you are right.
> 
> My nuxt folder does not contain any special character, but the parent folder does and it make nuxt error `some folder -> myprojectfolder -> nuxt-app`
> 
> Windows 10 Node v18.12.1 Npm 8.19.2

I can confirm. I fixed this issue renaming my main folder that contains space.

**Comment by manniL** [comment]
As mentioned already [in the comment above](https://github.com/nuxt/framework/issues/9074#issuecomment-1319389101) ☺️

**Comment by Julio-Campos-Swork** [comment]
El error es por que alguna ruta de tu directorio contiene un espacio, en mi caso lo tenia en una carpeta llamada "Documentos Julio" quitando el espacio compila de manera correcta

**Comment by J3r0github** [comment]
Indeed, happens if a space, or any "special letter" excluding "-" or "_" is present in the folder name.
An annoying bug at least.

**Comment by Feelixe-tin** [comment]
Maybe the priority should be upgraded from p3-minor-bug?
Since the bug can happen without you writing a single line of code, it will affect many users creating new projects on windows.
And having a space in a folder name of the project path is probably very common.

**Comment by jdgamble555** [comment]
Has the fix not been found yet? This seems like there should be a simple fix for this problem to still be there.

J

**Comment by manniL** [comment]
> Has the fix not been found yet? This seems like there should be a simple fix for this problem to still be there.

Anyone, including you, can feel free to investigate and send in a PR 😋

**Comment by jdgamble555** [comment]
**That's fair**, however, you guys could find this simple fix in a quarter of the time as one of its new users. I'm pretty sure it was from something a few PRs ago.  You guys made this **stable** knowing this problem exists, but I would bet this effects more than half of your users who are not going to decide to adopt it on their first use. 

> How often do people store their programming files in a folder with spaces or foreign characters? 

Honestly, no idea.

I am not currently using Nuxt, or I would be happy to help out. For me, this was my first go at it.

All that being said, this does look incredibly amazing, I thank you guys for the hard work, and I look forward to trying this out a few versions from now.

J

**Comment by yungsinatra0** [comment]
> This seems to be an issue when having "special characters" that have to be URL-encoded in the path. Could be umlauts like öäü, spaces and more.
> 
> @danielroe Could you add that to the issue?
> 
> Slightly related: #8706

Thank you for the workaround, worked like a charm! I was going crazy for the past day or so trying to figure out what was causing this issue.

**Comment by GlennHS** [comment]
Just to add another example to the pile:

**OS:** Windows 10 Home v21H2
**Nuxt Version:** Latest
**Error:** 500 [vite-node] [ERR_INVALID_URL_SCHEME] /node_modules/ufo/dist/index.mjs
**Directory:** C:\Users\Glenn Hamilton-Smith\Documents\Code\NuxtLayoutTest

Windows users can't easily change their user's root folder name without editing the registry and risking damaging the integrity of existing software on their machine (as most installed software not installed for all users uses your Windows user folder name at time of installation).

I'm nowhere near skilled enough to raise a PR or work on this issue but if I can assist at all with more info, error logs, screenshots etc please let me know! Weirdly Nuxt was working perfectly for me up until about a week or two ago so I'm not sure what's changed 🤔

**Comment by therealokoro** [comment]
@GlennHS i'll tell you what happened, you upgraded to RC.13 and above 😏....I've been stock in RC.12 for ages now 🥲🥲

**Comment by danielroe** [comment]
This should be resolved in ufo v1.0.1. which you can get with `npx nuxi upgrade`.

Let me know if not and I'll reopen.

**Comment by GlennHS** [comment]
As a note to future travellers to this page, I've found that using `npx nuxi upgrade` doesn't work for me, however installing a fresh Nuxt 3 project works perfectly fine. Not ideal, but given how this issue was only around for a brief time I doubt it's going to have much of an impact :)

@danielroe thank you so much for fixing this so quickly, I really appreciate it 😊
